lib/rules/no-floating-decimal.js

```diff
@@ -27,10 +29,18 @@ module.exports = {
 
                 if (typeof node.value === "number") {
                     if (node.raw.indexOf(".") === 0) {
-                        context.report(node, "A leading decimal point can be confused with a dot.");
+                        context.report({
+                            node,
+                            message: "A leading decimal point can be confused with a dot.",
+                            fix: fixer => fixer.insertTextBefore(node, "0")
+                        });
                     }
                     if (node.raw.indexOf(".") === node.raw.length - 1) {
-                        context.report(node, "A trailing decimal point can be confused with a dot.");
+                        context.report({
+                            node,
+                            message: "A trailing decimal point can be confused with a dot.",
+                            fix: fixer => fixer.insertTextAfter(node, "0")
```

Comment:
I'm not sure it is-- the rule is specifically about decimal points being confused for dot operators, I think, and so a decimal point in the middle of a numeric literal (even if it is on the trailing each of the mantissa) should be okay.